tasks/rh-sign-cosign-image/rh-sign-cosign-image.yaml

```diff
@@ -0,0 +1,104 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: rh-sign-cosign-image
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Task to sign container images in a Snapshot spec using the cosign.
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+    - name: secretName
+      description: Name of secret containing needed credentials 
+      type: string
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: sign-image
+      image: quay.io/konflux-ci/release-service-utils:38c3bfd7479c86b832cba5b61f9bbde40c469393
+      env:
+        - name: AWS_REGION
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_REGION
+        - name: AWS_ACCESS_KEY_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_ACCESS_KEY_ID
+        - name: AWS_SECRET_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_SECRET_ACCESS_KEY
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+        function fetch_cosign_key() {
+          kubectl get configmap --output json | jq -r ".data.SIG_KEY" | tr -d \"
+        } 
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        request=$(jq -r '.sign.request // "hacbs-signing-pipeline"' ${DATA_FILE})
+        CONFIG_MAP_NAME=$(jq -rc '.sign.configMapName // "signing-config-map"' ${DATA_FILE})
+
+        COMPONENTS_LENGTH=$(jq '.components |length' ${SNAPSHOT_PATH})
+
+        COSIGN_SIG_KEY=$(fetch_cosign_key $CONFIG_MAP_NAME)
+
+        for (( COMPONENTS_INDEX=0; COMPONENTS_INDEX<COMPONENTS_LENGTH; COMPONENTS_INDEX++ )); do
+            BUILD_CONTAINER_IMAGE=$(jq -r ".components[${COMPONENTS_INDEX}].containerImage" ${SNAPSHOT_PATH})
+            DIGEST="${BUILD_CONTAINER_IMAGE/*@}"
+
+            COMPONENT_NAME=$(jq -r ".components[${COMPONENTS_INDEX}].name" ${SNAPSHOT_PATH})
+            echo "Processing component ${COMPONENT_NAME}"
+            INTERNAL_CONTAINER_REF=$(jq -r ".components[${COMPONENTS_INDEX}].repository" ${SNAPSHOT_PATH})
+            INTERNAL_CONTAINER_IMAGE="${INTERNAL_CONTAINER_REF}@${DIGEST}"
+            PUB_CONTAINER_REF=$(translate-delivery-repo $INTERNAL_CONTAINER_REF | jq -rc ".[0]|.url" | tr -d \")
```

Comment:
The `tr` shoudln't be needed, right? Because you use `jq -r`, it will be the raw string without `""`. Also, the `-c` option is redundant in the `jq` - it's just a string, so it cannot be made any more compact. Also, the `|` is redundant as well. So I would suggest this:
```sh
PUB_CONTAINER_REF=$(translate-delivery-repo $INTERNAL_CONTAINER_REF | jq -r ".[0].url")
```

Reply:
Also, I was looking at what we do in rh-sign-image and I can see we sign the registry.access.redhat.com reference as well. I'm not even sure we could sign multiple times with cosign. And it's probably ok to just use the registry.redhat.io reference. But I have another question - in this case we only sign the container image referenced by digest, is that correct? Compare this to simple signing (rh-sign-image) where we sign every tag separately. I think the answer is that here we do want to sign just the digest reference. But I am not 100 % sure.

Reply:
> But I have another question - in this case we only sign the container image referenced by digest, is that correct? 
Yes, that's correct. Cosign has optional annotation where you can specify tag, but it's used only as optional attribute in the signature.
I wasn't sure how to identity works as it wasn't present in cosign before. But it looks like when you sign same digest with another identity it will just create another simplesigning.v1+json cosign blob. So code can be changed to sign multiple times for different registries

Reply:
Thanks. I can see this was done - at least when reviewing the follow up PR: https://github.com/konflux-ci/release-service-catalog/pull/467